### PR TITLE
Net core api

### PR DESCRIPTION
**Pull request by zkonstr**

### CODE REVIEW

File: DebtNote/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+namespace DebtNote.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UserController : ControllerBase
+    {
+        private readonly ApplicationContext _context;
+
+        public UserController(ApplicationContext context) => _context = context;
+        [HttpGet]
```
Comment (mrZmitser) [inline DebtNote/Controllers/UserController.cs:23]: Add spaces between the methods here and further

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
```
Comment (mrZmitser) [inline DebtNote/Database/Migrations/20220617120138_createdatabase.cs:3]: explain this

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace DebtNote.Database.Migrations
+{
+    public partial class createdatabase : Migration
```
Comment (mrZmitser) [inline DebtNote/Database/Migrations/20220617120138_createdatabase.cs]: what about classes name convention? rename to CreatedDatabase

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                columns: table => new
+                {
+                    Id = table.Column<int>(type: "int", nullable: false)
+                        .Annotation("SqlServer:Identity", "1, 1"),
+                    Money = table.Column<decimal>(type: "decimal(18,2)", nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_Payments", x => x.Id);
+                });
```
Comment (mrZmitser) [inline DebtNote/Database/Migrations/20220617120138_createdatabase.cs]: move the closing bracket `)` to the next line

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                {
+                    Id = table.Column<int>(type: "int", nullable: false)
+                        .Annotation("SqlServer:Identity", "1, 1"),
+                    Name = table.Column<string>(type: "nvarchar(max)", nullable: false),
+                    Money = table.Column<decimal>(type: "decimal(18,2)", nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_Users", x => x.Id);
+                });
```
Comment (mrZmitser) [inline DebtNote/Database/Migrations/20220617120138_createdatabase.cs]: and here

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropTable(
+                name: "Payments");
```
Comment (mrZmitser) [inline DebtNote/Database/Migrations/20220617120138_createdatabase.cs:44]: and here and further

File: DebtNote/Models/User.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace DebtNote.Models
+{
+    public class User
+    {
+        [Required]
+        public int Id { get; set; }
+        [Required]
+        public string? Name { get; set; }
```
Comment (mrZmitser) [inline DebtNote/Models/User.cs]: Why is name nullable?

File: DebtNote/Services/Implimentations/TransferService.cs [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        private IPaymentRepository<Payment>? Payments { get; set; }
+        private IUserRepository<User>? Users { get; set; }
+
+        public IUserRepository<User>? GetUsers()
+        {
+            return Users;
+        }
+
+        public void Transfer(int amount, IUserRepository<User>? users)
```
Comment (mrZmitser) [inline DebtNote/Services/Implimentations/TransferService.cs]: What the hell is going here? Why is it not in User Service? Why it receives a users repo?

File: DebtNote/Repositories/Implimentations/PaymentRepository.cs [hunk 9]
```diff
@@ -0,0 +1,33 @@
+﻿using DebtNote.Models;
+using DebtNote.Repositories.Interfaces;
+
+namespace DebtNote.Repositories.Implimentations
+{
+    public class PaymentRepository<Payment> : IPaymentRepository<Payment>
```
Comment (mrZmitser) [inline DebtNote/Repositories/Implimentations/PaymentRepository.cs]: What the sense of generics here?

File: DebtNote/Repositories/Implimentations/UserRepository.cs [hunk 10]
```diff
@@ -0,0 +1,39 @@
+﻿using DebtNote.Database;
+using DebtNote.Repositories.Interfaces;
+
+namespace DebtNote.Repositories.Implimentations
+{
+    public class UserRepository<User> : IUserRepository<User>
```
Comment (mrZmitser) [inline DebtNote/Repositories/Implimentations/UserRepository.cs]: What the sense of generics here?

File: DebtNote/Repositories/Implimentations/UserRepository.cs [hunk 11]
```diff
@@ -0,0 +1,39 @@
+﻿using DebtNote.Database;
+using DebtNote.Repositories.Interfaces;
+
+namespace DebtNote.Repositories.Implimentations
```
Comment (mrZmitser) [inline DebtNote/Repositories/Implimentations/UserRepository.cs]: implEmentation

File: DebtNote/Services/Implimentations/TransferService.cs [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        {
+
+            User user1 = users.Create(new User
```
Comment (mrZmitser) [inline DebtNote/Services/Implimentations/TransferService.cs]: why do you create users here, not in a higher level?

**Review by mrZmitser — CHANGES_REQUESTED** [review]
Reply to all the comments or fix stuff where required

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs (hunk 3 above)
Comment (zkonstr) [reply DebtNote/Database/Migrations/20220617120138_createdatabase.cs]: renamed

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs (hunk 2 above)
Comment (zkonstr) [reply DebtNote/Database/Migrations/20220617120138_createdatabase.cs:3]: I don't know anything about migrations, but it helps me to create and fill tables in DB by one command

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs (hunk 4 above)
Comment (zkonstr) [reply DebtNote/Database/Migrations/20220617120138_createdatabase.cs]: this file was auto-generated, but... Well, I'll move it

File: DebtNote/Models/User.cs (hunk 7 above)
Comment (zkonstr) [reply DebtNote/Models/User.cs]: To much warnings about it.

File: DebtNote/Services/Implimentations/TransferService.cs (hunk 8 above)
Comment (zkonstr) [reply DebtNote/Services/Implimentations/TransferService.cs]: It was just yesterday first try to use services. Now it's deleted

File: DebtNote/Services/Implimentations/TransferService.cs (hunk 12 above)
Comment (zkonstr) [reply DebtNote/Services/Implimentations/TransferService.cs]: just for test

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs (hunk 2 above)
Comment (mrZmitser) [reply DebtNote/Database/Migrations/20220617120138_createdatabase.cs:3]: it's not about migrations
https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs (hunk 3 above)
Comment (mrZmitser) [reply DebtNote/Database/Migrations/20220617120138_createdatabase.cs]: commit! :)

File: DebtNote/Database/Migrations/20220617120138_createdatabase.cs (hunk 4 above)
Comment (mrZmitser) [reply DebtNote/Database/Migrations/20220617120138_createdatabase.cs]: well, sorry, if it's generated do nothing here, leave as it was

**Pull request merged**